src/qengine/state/opencl.cpp

```diff
@@ -146,6 +146,237 @@ void QEngineOCL::Apply2x2(bitCapInt offset1, bitCapInt offset2, const complex* m
     }
 }
 
+bitLenInt QEngineOCL::Cohere(QEngineOCLPtr toCopy)
```

Comment:
Too bad this wasn’t able to leverage any of the existing generic fallout code, rather than adding another couple of api-specific staging routines.

Take another look at this and see if the code - either in OCL or here - can be normalized on a consistent calling convention to reduce some of the duplicate code between these various implementations.